AirCasting/SessionViews/EditViewModel.swift

```diff
@@ -0,0 +1,57 @@
+// Created by Lunar on 15/12/2021.
+//
+
+import Foundation
+import SwiftUI
+import CoreData
+
+class EditSessionViewModel: ObservableObject {
+    
+    @Published var sessionName = ""
+    @Published var sessionTags = ""
+    private let measurementStreamStorage: MeasurementStreamStorage
+    private let sessionUpdateService: SessionUpdateService
+
+    init(measurementStreamStorage: MeasurementStreamStorage, sessionUpdateService: SessionUpdateService) {
+        self.measurementStreamStorage = measurementStreamStorage
+        self.sessionUpdateService = sessionUpdateService
+    }
+    
+    func saveChanges(for uuid: SessionUUID, completion: @escaping () -> Void) {
```

Comment:
usually this `completion` pattern is not used inside viewmodels. If we want something to happen after it saves we should inform the view using some domain-level named property and use it instead. For example if we wanted an alert to be presented we should add something like:
```
@Published allChangesSavedAlertVisible: Bool = false
```
and here instead of calling completion we should set it to `true`.

This is because we don't really want views to decide on any non-view logic

Reply:
The only thing the view was deciding about here was dismissing the view which for me is view-logic. I changed this as suggested but I think the previous solution was a bit prettier 🙃